web_infrastructure/deploy_helper.py

```diff
@@ -337,6 +342,16 @@ def remove_unfinished_builds(self, releases_path):
 
         return changes
 
+    def remove_unfinished_link(self, path):
```

Comment:
Why have a procedure to remove the temporary link? Won't the temp link be removed during rename?

If you have a good reason for it to exist, I do think this could also support the _module.check_mode_
(i.e. not change anything, but know *if* something will change)